src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.inventory.demo.service.impl;
+
+import com.inventory.demo.dto.ProductDto;
+import com.inventory.demo.dto.Response;
+import com.inventory.demo.entity.Product;
+import com.inventory.demo.repository.ProductRepository;
+import com.inventory.demo.service.ProductService;
+import com.inventory.demo.util.ResponseBuilder;
+import org.modelmapper.Conditions;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+
+    private final ProductRepository productRepository;
+    private final ModelMapper modelMapper;
+
+    public ProductServiceImpl(ProductRepository productRepository, ModelMapper modelMapper) {
+        this.productRepository = productRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public Response saveProduct(ProductDto productDto) {
+        Product product = productRepository.findByBatchIdAndIsActive(productDto.getBatchId(), Boolean.TRUE);
```

Comment:
Can ignore the validation with batch. Apply product name validation only

Reply:
Removed the batch Id property and the validation